text_collector_examples/nvme_metrics.sh

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env bash
+set -eu
+
+# Dependencies: nvme-cli, jq (packages)
+# Based on code from
+# - https://github.com/prometheus/node_exporter/blob/master/text_collector_examples/smartmon.sh
+# - https://github.com/vorlon/check_nvme/blob/master/check_nvme.sh
+# Author: Henk <henk@wearespindle.com>
+
+output_format_awk="$(
+  cat <<'OUTPUTAWK'
+BEGIN { v = "" }
+v != $1 {
+  print "# HELP nvme_" $1 " SMART metric " $1;
+  print "# TYPE nvme_" $1 " gauge";
```

Comment:
I spoke too soon, I ran the output through `promtool check metrics` and it discovered an issue with the `_total` counters.

This should fix it:
```awk
  if ($1 ~ /_total$/)
    print "# TYPE nvme_" $1 " counter";
  else
    print "# TYPE nvme_" $1 " gauge";
```

Reply:
Added suggested fix.